### ISSUE TITLE
JDK 13: IllegalStateException: Couldn't find module modules of URI jrt:/.../

### ISSUE THREAD

**Issue by remal** [TNG/ArchUnit#303]
**ArchUnit version: 0.13.0.**

I'm having this exception: `IllegalStateException: Couldn't find module modules of URI jrt:/.../`. Instead `...` there is a random module from JDK.

It look like `JrtFileSystemProvider.getPath(URI)` method has been changed from this one in JDK 12:
```java
public Path getPath(URI uri) {
    // ...
    return getTheFileSystem().getPath(path);
    // ...
}
```
... to this one in JDK 13:
```java
public Path getPath(URI uri) {
    // ...
    return getTheFileSystem().getPath("/modules" + path);
    // ...
}
```

This is the reason why `com.tngtech.archunit.core.importer.ModuleLocationFactory$ModuleLocation.parseModuleName(Path, NormalizedUri)` method fails. It seems the method uses only the first part of a path of the URI, which has been changed from `/jdk.zipfs` (for example) in JDK 12 to `/modules/jdk.zipfs` in JDK 13.

The stacktrace looks like this:
```
Couldn't find module modules of URI jrt:/jdk.jfr/
java.lang.IllegalStateException: Couldn't find module modules of URI jrt:/jdk.jfr/
	at com.tngtech.archunit.thirdparty.com.google.common.base.Preconditions.checkState(Preconditions.java:721)
	at com.tngtech.archunit.core.importer.ModuleLocationFactory$ModuleLocation.parseModuleName(ModuleLocationFactory.java:87)
	at com.tngtech.archunit.core.importer.ModuleLocationFactory$ModuleLocation.<init>(ModuleLocationFactory.java:67)
	at com.tngtech.archunit.core.importer.ModuleLocationFactory.create(ModuleLocationFactory.java:45)
	at com.tngtech.archunit.core.importer.Location.of(Location.java:168)
	at com.tngtech.archunit.core.importer.Location.of(Location.java:160)
	at com.tngtech.archunit.core.importer.Locations.of(Locations.java:62)
	at com.tngtech.archunit.core.importer.Locations.archiveLocationsOf(Locations.java:164)
	at com.tngtech.archunit.core.importer.Locations.findMissedClassesDueToLackOfPackageEntry(Locations.java:146)
	at com.tngtech.archunit.core.importer.Locations.getResourceLocations(Locations.java:123)
	at com.tngtech.archunit.core.importer.Locations.getLocationsOf(Locations.java:116)
	at com.tngtech.archunit.core.importer.Locations.ofPackage(Locations.java:76)
	at com.tngtech.archunit.junit.ClassCache$RequestedLocations$Specific.locationsOf(ClassCache.java:220)
	at com.tngtech.archunit.junit.ClassCache$RequestedLocations$Specific.getLocationsOfPackages(ClassCache.java:187)
	at com.tngtech.archunit.junit.ClassCache$RequestedLocations$Specific.<init>(ClassCache.java:177)
	at com.tngtech.archunit.junit.ClassCache$RequestedLocations$Specific.<init>(ClassCache.java:170)
	at com.tngtech.archunit.junit.ClassCache$RequestedLocations.by(ClassCache.java:234)
	at com.tngtech.archunit.junit.ClassCache.getClassesToAnalyzeFor(ClassCache.java:76)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.lambda$createChildren$1(ArchUnitTestDescriptor.java:82)
	at com.tngtech.archunit.thirdparty.com.google.common.base.Suppliers$MemoizingSupplier.get(Suppliers.java:120)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor$ArchUnitRuleDescriptor.execute(ArchUnitTestDescriptor.java:159)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor$ArchUnitRuleDescriptor.execute(ArchUnitTestDescriptor.java:142)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at ...
	at ...
	at ...
	at ...
```

**Comment by codecholeric** [comment]
Snap, you're absolutely right, thanks for reporting this! :frowning: 
Back then when I implemented this, I thought relying on `java.nio.file.Path` would make the whole thing more resilient than low level URI processing, now it turns out exactly the opposite is true :frowning_face: Didn't expect the implementation to change there. (but I have to admit, even though reading JEP-220 in detail back then, I relied on my implementation tests of JDK 9 instead of questioning why the spec for the JRT file system has not been followed with that respect).

We should also work on the CI to always include a Maven integration tests for any new JDK version we support (relying on ASM has made us a little too confident, I think, because I've noticed that we haven't added JDK 12, 13 or 14 to the list of JDKs to run Maven integration tests against).

I think I've fixed this with #304, but I'm on holiday at the moment, thus I can only release the fix when I'm back on Tuesday, sorry for any inconvenience this causes! (since it's only 2 work days, I guess it's not worth it to find someone else to release the fix...)
